### PR TITLE
Np 22111 doi registrar client

### PR DESCRIPTION
**Pull request by ketilaa**
Added support for deleting draft DOI independently of draft publication being marked for deletion

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -188,11 +186,51 @@ Resources:
             Type: SQS
             Destination: !GetAtt DataCiteDraftDoiHandlerDLQ.Arn
 
-  DeleteDraftDoiHandler:
+  DeleteDraftDoiHandlerFunction:
```
[the remark this answers is not in the document]
Comment (torarnet) [reply template.yaml:189]: I mean, being referenced in OpenApi

[the remark this answers is not in the document]
Comment (ketilaa) [reply template.yaml:189]: These are functions that are being triggered by events in EventBridge, so they are not exposed through API Gateway.

File: datacite-delete-draft-doi-handler/src/main/java/no/unit/nva/datacite/handlers/doi/DeleteDraftDoiHandler.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Override
+    protected DoiUpdateEvent processInputPayload(DoiUpdateRequestEvent input,
+                                                 AwsEventBridgeEvent<AwsEventBridgeDetail<DoiUpdateRequestEvent>> event,
+                                                 Context context) {
+
+        var doi = verifyEventHasDoi(input);
+
+        var customerId = input.getItem().getPublisher().getId();
```
Comment (torarnet) [inline datacite-delete-draft-doi-handler/src/main/java/no/unit/nva/datacite/handlers/doi/DeleteDraftDoiHandler.java:53]: item never null?

**Review by torarnet — APPROVED** [review]

File: datacite-delete-draft-doi-handler/src/main/java/no/unit/nva/datacite/handlers/doi/DeleteDraftDoiHandler.java (hunk 2 above)
Comment (ketilaa) [reply datacite-delete-draft-doi-handler/src/main/java/no/unit/nva/datacite/handlers/doi/DeleteDraftDoiHandler.java:53]: Jeg fikser denne.

**Review by torarnet — APPROVED** [review]

**Pull request merged**